src/toil/batchSystems/kubernetes.py

```diff
@@ -205,11 +218,15 @@ def _api(self, kind, max_age_seconds = 5 * 60):
                 
         else:
             # We need an API object
+            # WIP -------------------------------------------------------------------- 
+            for attempt in retry_http(predicate=handle_retriable_exception):
+                with attempt:
+                    api_get = self._apis[kind]
```

Comment:
This is wrapping the dict access in the retry, not any actual use of the API object.

But we could turn `_api()` into a context manager, and make everything that uses a Kubernetes connection be inside a `with self._api('whatever'):`, and let `_api()` handle all the retries maybe?

Or you could apply this `with attempt in retry_http...` to all the places where we call any methods on the PAI objects we get from `_api()`.